README.rdoc

```diff
@@ -0,0 +1,54 @@
+= LRUHash
+
+This is a hash class that holds a maximum number of elements.  If adding an element causes the hash to grow over this size, the least recently used element is removed. Elements in the hash are maintained in LRU order.
+
+Useful for implementing in-process caches where memory growth control is desired.
+
+
+Example:
+
+  myhash = LRUHash.new
+  myhash.max = 2
+
+  myhash[1] = 1
+  myhash[2] = 2
+
+  myhash.length # => 2
+
+  myhash[3] = 3
+
+  myhash.length # => 2
+
+  myhash[1] # => nil
```

Comment:
@primiti  Can you make the example use different keys and values? It's confusing when 1 is both! And ideally neither is an integer, since you're showing a key, a value, and a length all of which are `2`!